### PR TITLE
XRay bugfix, Timer improvement and more

### PR DESCRIPTION
**Pull request by FDPUser443**
There are some new things:

XRay:
 - fixed gamma level not resetting after disabling XRay [Should I have used onEnable and onDisable methods instead of  two "if" statements ? =)) ]

AutoArmor: 
 - SwitchBackLegit option is now visible only when Hotbar option is activated

Timer:
 - Added Timing option with few options ( https://www.youtube.com/watch?v=UAqT0VunA9o )

HitBox:
 - Entity hitboxes size are depend on HitBox module now (When F3 + B is pressed) ( https://www.youtube.com/watch?v=hEK88y7hCwo )

### CODE REVIEW

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/render/XRay.kt [hunk 1]
```diff
@@ -47,7 +47,15 @@ object XRay : Module("XRay", ModuleCategory.RENDER) {
...
+    private var prevGammaLevel = 0f
+
     override fun onToggle(state: Boolean) {
+        if (state) {
+            prevGammaLevel = mc.gameSettings.gammaSetting
+        }
         mc.renderGlobal.loadRenderers()
+        if (!state) {
+            mc.gameSettings.gammaSetting = prevGammaLevel
+        }
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/features/module/modules/render/XRay.kt]: Yeah onEnable and onDisable would be better.

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Timer.kt [hunk 2]
```diff
@@ -11,13 +11,13 @@ import net.ccbluex.liquidbounce.event.WorldEvent
...
 import net.ccbluex.liquidbounce.utils.MovementUtils.isMoving
-import net.ccbluex.liquidbounce.value.BoolValue
 import net.ccbluex.liquidbounce.value.FloatValue
+import net.ccbluex.liquidbounce.value.ListValue
 
 object Timer : Module("Timer", ModuleCategory.WORLD) {
 
     private val speed by FloatValue("Speed", 2F, 0.1F..10F)
-    private val onMove by BoolValue("OnMove", true)
+    private val timing by ListValue("Timing", arrayOf("OnMove", "NoMove", "Both"), "OnMove")
```
Comment (CzechHek) [inline src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Timer.kt]: I would personally find it better to call it "Mode" with options as "OnMove", "NoMove", "Always".

(it doesn't have to do anything with timing and "Both" is misleading if you are setting Timer values from chat)

**Review by CzechHek — CHANGES_REQUESTED** [review]

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Timer.kt (hunk 2 above)
Comment (gabrielvicenteYT) [reply src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Timer.kt]: +1

**Comment by FDPUser443** [comment]
Thanks for your valuable review;

**Comment by ghost** [comment]
We good? Anything else to update/change before this gets merged?

**Comment by FDPUser443** [comment]
Oh, I need to update it according to #1220

**Comment by FDPUser443** [comment]
Fixed HitBox not working properly and updated hitbox rendering when F3 + B is pressed ( https://streamable.com/u8ivhc )

**Comment by ghost** [comment]
Great.

**Pull request merged**